plugins/inputs/cisco_telemetry_mdt_netconf/cisco_telemetry_mdt_netconf.go

```diff
@@ -0,0 +1,930 @@
+}
+
+// notificationSubscriptionRequest given as input to the plugin
+type notificationSubscriptionRequest struct {
+	Stream string   `toml:"stream"`
+	Tags   []string `toml:"tags"`
+}
+
+// getRequest given as input to the plugin
+type getRequest struct {
+	SelectFilter string          `toml:"xpath_filter"`
+	Period       config.Duration `toml:"period"`
+	Tags         []string        `toml:"tags"`
+}
+
+// netconfSubscriptionRequest sent through NETCONF, as per specification
+type netconfSubscriptionRequest struct {
+	XMLName         xml.Name `xml:"urn:ietf:params:xml:ns:yang:ietf-event-notifications establish-subscription"`
+	YangPush        string   `xml:"xmlns:yp,attr"`
+	Stream          string   `xml:"stream"`
+	XPathFilter     string   `xml:"urn:ietf:params:xml:ns:yang:ietf-yang-push xpath-filter"`
+	Period          uint64   `xml:"urn:ietf:params:xml:ns:yang:ietf-yang-push period,omitempty"`           // hundreds of a second 100=1s
+	DampeningPeriod *uint64  `xml:"urn:ietf:params:xml:ns:yang:ietf-yang-push dampening-period,omitempty"` // on-change
+}
+
+// netconfYangPush notification received through NETCONF, as per specification
+type netconfYangPush struct {
+	netconf.Notification
+	PushUpdate struct {
+		Content struct {
+			InnerXML []byte `xml:",innerxml"`
+		} `xml:"datastore-contents-xml"`
+	} `xml:"urn:ietf:params:xml:ns:yang:ietf-yang-push push-update"`
+}
+
+// netconfYangEvent notification received through NETCONF, as per specification
+type netconfYangEvent struct {
+	netconf.Notification
+	InnerXML []byte `xml:",innerxml"`
+}
+
+// CiscoTelemetryNETCONF plugin
+type CiscoTelemetryNETCONF struct {
+	ServerAddress string `toml:"server_address"`
+
+	Dsrs *dialinSubscriptionRequestsService `toml:"subscription_service"`
+	Grs  *getRequestsService                `toml:"get_service"`
+
+	// SSH connection - credentials
+	Username string `toml:"username"`
+	Password string `toml:"password"`
+
+	// SSH connection - security
+	IgnoreServerAuthenticity bool   `toml:"ignore_server_authenticity"`
+	ServerPublicKey          string `toml:"server_public_key"`
+
+	// Xpaths and tags of all operations
+	userXpaths map[string]interface{}
+	userTags   map[string]interface{}
+
+	Redial config.Duration `toml:"redial"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	acc    telegraf.Accumulator
+	cancel context.CancelFunc
+
+	// Waitgroup for monitoring the main services
+	waitgroup sync.WaitGroup
+}
+
+// Init ensures the plugin is configured correctly.
+func (c *CiscoTelemetryNETCONF) Init() error {
+	if c.ServerAddress == "" {
+		return fmt.Errorf("server address cannot be empty")
+	}
+	if c.Username == "" || c.Password == "" {
+		return fmt.Errorf("ssh credentials cannot be empty")
+	}
+	if !c.IgnoreServerAuthenticity && c.ServerPublicKey == "" {
+		return fmt.Errorf("public key must exist when ignore server authenticity is disabled")
+	}
+	if time.Duration(c.Redial).Seconds() < 1 {
+		return fmt.Errorf("redial must be greater than or equal to 1s")
+	}
```

Comment:
If there are no errors please use `errors.New()`
```suggestion
	if c.ServerAddress == "" {
		return errors.New("server address cannot be empty")
	}
	if c.Username == "" || c.Password == "" {
		return errors.New("ssh credentials cannot be empty")
	}
	if !c.IgnoreServerAuthenticity && c.ServerPublicKey == "" {
		return errors.New("public key must exist when ignore server authenticity is disabled")
	}
	if time.Duration(c.Redial).Seconds() < 1 {
		return errors.New("redial must be greater than or equal to 1s")
	}
```